pkg/util/ftp/auth_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftp
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+)
+
+const (
+	informerDefaultResync = 30 * time.Second
+
+	testKeyString   = "it is a valid API key"
+	testTokenString = "it is a valid token"
+)
+
+type secretConfig struct {
+	name string
+	data map[string][]byte
+}
+
+func prepareSecrets(ns string, secretConfigs []secretConfig) []*corev1.Secret {
+	secrets := make([]*corev1.Secret, 0)
+	for _, s := range secretConfigs {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      s.name,
+				Namespace: ns,
+			},
+			Data: s.data,
+		}
+		secrets = append(secrets, secret)
+	}
+	return secrets
+}
+
+type testClient struct {
+	client          kubernetes.Interface
+	informerFactory informers.SharedInformerFactory
+}
+
+func (c *testClient) start(stopCh <-chan struct{}) {
+	c.informerFactory.Start(stopCh)
+}
+
+func (c *testClient) waitForSync(stopCh <-chan struct{}) {
+	c.informerFactory.WaitForCacheSync(stopCh)
+}
+
+func newTestClient(coreObjects []runtime.Object, crdObjects []runtime.Object) *testClient {
+	client := fake.NewSimpleClientset(coreObjects...)
+	return &testClient{
+		client:          client,
+		informerFactory: informers.NewSharedInformerFactory(client, informerDefaultResync),
+	}
+}
```

Comment:
I don't think you need these codes, you just need a fake client `fake.NewSimpleClientset(coreObjects...)`,and pass it to `ParseFileServerAuth`.

Reply:
yeah ,it works